From: Yu Watanabe <watanabe.yu+github@gmail.com>
Date: Mon, 9 Sep 2019 01:27:56 +0900
Subject: udev: add missing flag for OPTIONS=static_node

Fixes RHBZ#1740664.

(cherry picked from commit 5abba26e2f9eb9336e701228220f0ef62c95f316)
(cherry picked from commit 75a2845e5a193aeb120dbccd0da0f174230b6d71)
---
 src/udev/udev-rules.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/udev/udev-rules.c b/src/udev/udev-rules.c
index 1642f10..21f84b2 100644
--- a/src/udev/udev-rules.c
+++ b/src/udev/udev-rules.c
@@ -499,6 +499,9 @@ static int rule_line_add_token(UdevRuleLine *rule_line, UdevRuleTokenType type,
                         TK_A_OWNER_ID, TK_A_GROUP_ID, TK_A_MODE_ID))
                 SET_FLAG(rule_line->type, LINE_HAS_DEVLINK, true);
 
+        else if (token->type == TK_A_OPTIONS_STATIC_NODE)
+                SET_FLAG(rule_line->type, LINE_HAS_STATIC_NODE, true);
+
         else if (token->type >= _TK_A_MIN ||
                  IN_SET(token->type, TK_M_PROGRAM,
                         TK_M_IMPORT_FILE, TK_M_IMPORT_PROGRAM, TK_M_IMPORT_BUILTIN,
